python/GafferUI/ColorChooser.py

```diff
@@ -163,16 +163,32 @@ def _displayTransformChanged( self ) :
 		GafferUI.Slider._displayTransformChanged( self )
 		self._qtWidget().update()
 
+class _ColorFieldWidget( QtWidgets.QWidget ) :
+
+	def __init__( self, parent = None ) :
+
+		QtWidgets.QWidget.__init__( self, parent )
+
+		self.__size = 216
+
+	def resizeEvent( self, event ) :
+
+		w = event.size().width()
+		h = event.size().height()
+
+		if w != h :
+			self.__size = h
+			self.setMinimumWidth( h )
```

Comment:
Yes, this will be doing a double layout (or maybe even more depending on how the constraints work out?) before `w == h` and it can stop.

The closest I've found to a canonical way of keeping something square in Qt is to use `heightForWidth()` where it returns the input width. We'd need something like `widthForHeight()` since we know the height (the total height of the sliders) and need to figure out the width.

I think the most correct solution would probably be to create a custom layout.

Reply:
In https://github.com/GafferHQ/gaffer/pull/6003/commits/dea359b3080ab8442e655d29bcc753d24aff43bd, I implemented a custom layout specific to the row for the color field and it's working well.

I tried a couple alternatives to avoid the complexity of a custom layout, but they were not as successful. One was listening for the resize event on the slider container and resizing the color field at that time. But it ran into pretty heavy flickering issues on the first layout.